api/envoy/api/v2/route/route_components.proto

```diff
@@ -1568,6 +1568,15 @@ message HeaderMatcher {
     //
     // * The suffix *abcd* matches the value *xyzabcd*, but not for *xyzbcd*.
     string suffix_match = 10 [(validate.rules).string = {min_bytes: 1}];
+
+    // If specified, header match will be performed based on whether the header value contains
```

Comment:
I wonder if we just want to add support general globs, rather than special casing this.

Reply:
@htuch This is being added for issue opened here: https://github.com/envoyproxy/envoy/issues/12590

This is no more a special case than `suffix-match` or `present-match`. @htuch and @mattklein123 Later today I will add the same matcher for StringMatcher as well.  Please let me know if we want to proceed this way.

Is there another PR or design doc available for wildcard based matches? Regex matching with `std::regex` can cause catastrophic backtracking and occasional crashes. Is there such a possibility with glob/wildcard matching too? Substring matches are O(n) as they are based on KMP.

Reply:
@htuch I did a little more digging. I see that the `Xray` tracer extension has a wildcard match:
https://github.com/envoyproxy/envoy/pull/8526/files#diff-620892d553c2be71cf26761d0c9db408R26

I am still breaking it down but right now it looks like it supports `?` and `*` which is good enough for our case. 

Do you want just these two wildcards or a full set? If you're okay with this I can refactor the code a little to reuse the same matcher.

Reply:
I think we would want a syntax more like https://github.com/envoyproxy/envoy/issues/7763. Maybe we should go with a simple contains for now.